demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -0,0 +1,841 @@
+        pack_error_ignore_list = self.get_error_ignore_list(os.path.basename(pack_path))
+
+        valid_pack.add(self.validate_pack_unique_files(pack_path, pack_error_ignore_list))
+
+        for content_dir in os.listdir(pack_path):
+            content_entity_path = os.path.join(pack_path, content_dir)
+            if content_dir in CONTENT_ENTITIES_DIRS:
+                valid_pack.add(self.run_validation_on_content_entities(content_entity_path, pack_error_ignore_list))
+
+        return all(valid_pack)
+
+    def run_validation_on_content_entities(self, content_entity_dir_path, pack_error_ignore_list):
+        """Gets non-pack folder and runs validation within it (Scripts, Integrations...)
+
+        Returns:
+            bool. true if all files in directory are valid, false otherwise.
+        """
+        valid_directory = set()
+        for file_name in os.listdir(content_entity_dir_path):
+            file_path = os.path.join(content_entity_dir_path, file_name)
+            if os.path.isfile(file_path):
+                if file_path.endswith('.json') or file_path.endswith('.yml') or file_path.endswith('.md'):
+                    valid_directory.add(self.run_validations_on_file(file_path, pack_error_ignore_list))
+
+            else:
+                valid_directory.add(self.run_validation_on_package(file_path, pack_error_ignore_list))
+
+        return all(valid_directory)
+
+    def run_validation_on_package(self, package_path, pack_error_ignore_list):
+        valid_package = set()
+
+        for file_name in os.listdir(package_path):
+            file_path = os.path.join(package_path, file_name)
+            if file_path.endswith('.yml') or file_path.endswith('.md'):
+                valid_package.add(self.run_validations_on_file(file_path, pack_error_ignore_list))
+
+        return all(valid_package)
+
+    # flake8: noqa: C901
+    def run_validations_on_file(self, file_path, pack_error_ignore_list, is_modified=False,
+                                old_file_path=None, modified_files=None, added_files=None):
+        """Choose a validator to run for a single file. (i)
+
+        Args:
+            modified_files: A set of modified files - used for RN validation
+            added_files: A set of added files - used for RN validation
+            old_file_path: The old file path for renamed files
+            pack_error_ignore_list: A dictionary of all pack ignored errors
+            file_path: the file on which to run.
+            is_modified: whether the file is modified or added.
+
+        Returns:
+            bool. true if file is valid, false otherwise.
+        """
+
+        file_type = find_type(file_path)
+
+        if file_type in SKIPPED_FILE_TYPES:
+            return True
+
+        elif file_type is None:
+            error_message, error_code = Errors.file_type_not_supported()
+            if self.handle_error(error_message=error_message, error_code=error_code, file_path=f'\n{file_path}'):
```

Comment:
why `\n` ?

Reply:
it looks like a patch, let's come up with another solution please, this is a re-write we can and should make it look good